src/PolygonWorldID.sol

```diff
@@ -51,6 +58,13 @@ contract PolygonWorldID is WorldIDBridge, FxBaseChildTunnel, Ownable {
         override
         validateSender(sender)
     {
+        address(this).call(message);
```

Comment:
I think you want to pull out the selector from the `message`, and validate that it matches either the `receiveRoot.selector` or `receiveRootHistoryExpiry.selector`. This way, you guard against potentially malformed messages.

Reply:
will do

Reply:
although on another thought idk if you can decode `encodeWithSignature`